[2024-03-25T20:23:40.947+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Book-Recommendation-DAG.do_clean_to_parquet_task manual__2024-03-25T20:23:32.942899+00:00 [queued]>
[2024-03-25T20:23:40.952+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Book-Recommendation-DAG.do_clean_to_parquet_task manual__2024-03-25T20:23:32.942899+00:00 [queued]>
[2024-03-25T20:23:40.952+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 3
[2024-03-25T20:23:40.958+0000] {taskinstance.py:2217} INFO - Executing <Task(PythonOperator): do_clean_to_parquet_task> on 2024-03-25 20:23:32.942899+00:00
[2024-03-25T20:23:40.961+0000] {standard_task_runner.py:60} INFO - Started process 1328 to run task
[2024-03-25T20:23:40.964+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'Book-Recommendation-DAG', 'do_clean_to_parquet_task', 'manual__2024-03-25T20:23:32.942899+00:00', '--job-id', '259', '--raw', '--subdir', 'DAGS_FOLDER/book-recommendation-dag.py', '--cfg-path', '/tmp/tmpqhs0njxe']
[2024-03-25T20:23:40.965+0000] {standard_task_runner.py:88} INFO - Job 259: Subtask do_clean_to_parquet_task
[2024-03-25T20:23:40.988+0000] {task_command.py:423} INFO - Running <TaskInstance: Book-Recommendation-DAG.do_clean_to_parquet_task manual__2024-03-25T20:23:32.942899+00:00 [running]> on host 992e1288d09b
[2024-03-25T20:23:41.023+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='iamraphson' AIRFLOW_CTX_DAG_ID='Book-Recommendation-DAG' AIRFLOW_CTX_TASK_ID='do_clean_to_parquet_task' AIRFLOW_CTX_EXECUTION_DATE='2024-03-25T20:23:32.942899+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-03-25T20:23:32.942899+00:00'
[2024-03-25T20:23:41.624+0000] {book-recommendation-dag.py:106} INFO - Done cleaning up!
[2024-03-25T20:23:42.602+0000] {book-recommendation-dag.py:106} INFO - Done cleaning up!
[2024-03-25T20:23:43.577+0000] {warnings.py:109} WARNING - /opt/***/dags/book-recommendation-dag.py:52: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  dataset_df = pd.read_csv(f'{dataset_download_path}{filename}')

[2024-03-25T20:23:43.780+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/book-recommendation-dag.py", line 71, in do_clean_to_parquet
    dataset_df = dataset_df.reset_index(level=2, drop=True)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/frame.py", line 6162, in reset_index
    level = [self.index._get_level_number(lev) for lev in level]
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/frame.py", line 6162, in <listcomp>
    level = [self.index._get_level_number(lev) for lev in level]
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 1915, in _get_level_number
    self._validate_index_level(level)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 1906, in _validate_index_level
    raise IndexError(
IndexError: Too many levels: Index has only 1 level, not 3
[2024-03-25T20:23:43.803+0000] {taskinstance.py:1149} INFO - Marking task as UP_FOR_RETRY. dag_id=Book-Recommendation-DAG, task_id=do_clean_to_parquet_task, execution_date=20240325T202332, start_date=20240325T202340, end_date=20240325T202343
[2024-03-25T20:23:43.816+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 259 for task do_clean_to_parquet_task (Too many levels: Index has only 1 level, not 3; 1328)
[2024-03-25T20:23:43.838+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-03-25T20:23:43.859+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
